client/test/backend.go

```diff
@@ -175,11 +176,12 @@ func (b *MockBackend) Progress(_ context.Context, req channel.ProgressReq) error
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
+	timeout := time.Now().Add(time.Duration(req.Params.ChallengeDuration) * time.Millisecond)
```

Comment:
Shouldn't the ChallengeDuration be given in seconds?

Reply:
Yes, but this would be too long for the tests. Here we scale down to milliseconds.

Reply:
Okay, makes sense.